src/Blockcore.Indexer.Core/Storage/Mongo/MongoData.cs

```diff
@@ -680,6 +680,29 @@ public QueryAddress AddressBalance(string address)
          };
       }
 
+
+      /// <summary>
+      /// Calculates the balance for specified list of addresses.
+      /// </summary>
+      /// <param name="addresses"></param>
+      public List<AddressBalanceResult> MultipleAddressBalances(List<string> addresses)
+      {
+         var result = new ConcurrentBag<AddressBalanceResult>();
+
+         Parallel.ForEach(addresses, address =>
+         {
+            result.Add(new AddressBalanceResult()
+            {
+               Address = address,
+               Balance = ComputeAddressBalance(address).Available
```

Comment:
Isn't it better to sum the outputs in the UTXO table? can be much faster and easy to implement.